### DISCUSSION TITLE
[Feature] Allow using same metro bundler for native and web

### DISCUSSION THREAD

**Discussion by jmatsushita** [tamagui/tamagui#1361] [General]
Hi there,

Currently the bundler needs to be started with a different `TAMAGUI_TARGET` however these days, the tooling is converging and the same bundler can be used for bundling for native and web projects with the [web bundling features of metro](https://docs.expo.dev/guides/customizing-metro/#web-support).

It would be great if it was possible to pickup the `TAMAGUI_TARGET` dynamically depending of whether there's a web bundling request or an ios/android  bundling request.

I suspect it's much easier said than done, but at least it's said now ;)

Cheers,

Jun

**Comment by LunatiqueCoder** [comment]
Hello. I managed to fix this by starting two bundlers, one on **8081** for native and one on **8082** for web.

**Comment by jmatsushita** [comment]
Thanks for you comment @LunatiqueCoder that's also what I do, but my hope is that we could use the same bundler instead 😄

**Comment by senghuotlay** [comment]
but do u guys have to re-configure webpack to make it work, or we can just run via expo-web?

**Comment by jmatsushita** [comment]
These days, using only one metro bundler process seems to work just fine (and seems to not even require `TAMAGUI_TARGET` anymore c.f. https://github.com/tamagui/tamagui/discussions/2243)

@senghuotlay I'm using the metro bundler rather than webpack.
